src/ImageSharp.Web/Resolvers/FileProviderImageResolver.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileProviders;
+
+namespace SixLabors.ImageSharp.Web.Resolvers
+{
+    /// <summary>
+    /// Provides means to manage image buffers from an <see cref="IFileInfo"/> instance.
+    /// </summary>
+    public sealed class FileProviderImageResolver : IImageResolver
+    {
+        private readonly IFileInfo fileInfo;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="FileProviderImageResolver"/> class.
+        /// </summary>
+        /// <param name="fileInfo">The file info.</param>
+        public FileProviderImageResolver(IFileInfo fileInfo) => this.fileInfo = fileInfo;
```

Comment:
Should this also do a null check?
```suggestion
        public FileProviderImageResolver(IFileInfo fileInfo) => this.fileInfo = fileInfo ?? throw new ArgumentNullException(nameof(fileInfo));;
```

Reply:
I wouldn't bother. The default implementation is smart enough to only call this type when the `IFileInfo ` is not null. If someone doesn't do this with their custom code then it's up to them.